dm/dm/master/server.go

```diff
@@ -2641,6 +2641,67 @@ func (s *Server) sharedLogic(ctx context.Context, req interface{}, respPointer i
 	return true
 }
 
+func (s *Server) checkStartValidationParams(req *pb.StartValidationRequest) (
+	map[string]map[string]config.SubTaskConfig, string, bool,
+) {
+	explicitModeOrStartTime := req.Mode != nil || req.StartTime != nil
+	if req.Mode != nil {
+		mode := req.GetModeValue()
+		if mode != config.ValidationFull && mode != config.ValidationFast {
+			msg := fmt.Sprintf("validation mode should be either `%s` or `%s`",
+				config.ValidationFull, config.ValidationFast)
+			return nil, msg, false
+		}
+	}
+
+	startTime := req.GetStartTimeValue()
+	if startTime != "" {
+		if _, err := utils.ParseStartTime(startTime); err != nil {
+			return nil, "start-time should be in the format like '2006-01-02 15:04:05' or '2006-01-02T15:04:05'", false
+		}
+	}
+
+	subTaskCfgs := s.scheduler.GetSubTaskCfgsByTaskAndSource(req.TaskName, req.Sources)
+	if len(subTaskCfgs) == 0 {
+		if req.TaskName == "" {
+			return nil, fmt.Sprintf("fail to get subtask config by sources `%v`", req.Sources), false
+		}
+		return nil, fmt.Sprintf("fail to get subtask config by task name `%s` and sources `%v`",
+			req.TaskName, req.Sources), false
+	}
+
+	var enabledValidator string
+	allEnabled, noneEnabled := true, true
+	for taskName := range subTaskCfgs {
+		for sourceID := range subTaskCfgs[taskName] {
+			if s.scheduler.ValidatorEnabled(taskName, sourceID) {
+				enabledValidator = taskName + " with source " + sourceID
+				noneEnabled = false
+			} else {
+				allEnabled = false
+			}
+		}
+	}
+
+	// if user set mode or start-time explicitly, then we do start operation.
+	// none of the target validators should have enabled
+	if explicitModeOrStartTime && !noneEnabled {
+		msg := fmt.Sprintf("some of target validator(%s) has already enabled.", enabledValidator)
+		if allEnabled {
+			msg = "all target validator has enabled, cannot do 'validation start' with explicit mode or start-time"
+		}
+		return nil, msg, false
+	}
+	// no explicit mode or start-time:
+	// - do resume when all have enabled
+	// - do start when none enabled
+	if !explicitModeOrStartTime && !allEnabled && !noneEnabled {
+		return nil, fmt.Sprintf("some of target validator(%s) has already enabled.", enabledValidator), false
```

Comment:
the final message which is displayed to user can be more clear, such as "some (%s) but not all validator has already enabled, so we can't start nor resume them in one command" or something